octopoes/octopoes/repositories/ooi_repository.py

```diff
@@ -296,9 +305,21 @@ def list_oois(
         limit: int = 20,
         scan_levels: set[ScanLevel] = DEFAULT_SCAN_LEVEL_FILTER,
         scan_profile_types: set[ScanProfileType] = DEFAULT_SCAN_PROFILE_TYPE_FILTER,
+        search_string: str | None = None,
+        order_by: Literal["scan_level", "object_type"] = "object_type",
+        asc_desc: Literal["asc", "desc"] = "asc",
     ) -> Paginated[OOI]:
         types = to_concrete(types)
 
+        search_statement = (
+            f"""[?e :xt/id ?id]
+                                [(clojure.string/includes? ?id \"{escape_string(search_string)}\")]"""
```

Comment:
The escaping results in an XTDB error, for example try searching for `|`. We also don't seem to handle the error from octopoes correctly, because I get an exception in rocky too.

If I do this manually I get the error `{"error": "Malformed \"application/edn\" request."}`. As far as I can see the problem is that we can't just have arbitrary escape sequences in strings, only valid ones. Given that this is just a string with no other meaning, I think we should only escape `"` and `\` itself.

Reply:
Like this?